dapp/src/components/layout.js

```diff
@@ -24,7 +29,39 @@ const Layout = ({
   medium,
   isStakePage,
   showUniswapNotice,
+  storeTransaction,
+  storeTransactionError,
 }) => {
+  const { connector, account } = useWeb3React()
+  const [showRebaseOptInNotice, setShowRebaseOptInNotice] = useState(false)
+  const ousdContract = useStoreState(ContractStore, (s) =>
+    get(s, 'contracts.ousd')
+  )
+
+  useEffect(() => {
+    async function checkRebaseState() {
+      if (ousdContract && account) {
+        const isSafe = !!get(connector, 'safe.safeAddress', false)
+        const rebaseOptInState = await ousdContract.rebaseState(account)
```

Comment:
BalanceHeader runs very similar code to this one. Should this checking logic be moved somewhere else (like `AccountListener` for example) and the result stored in Account or Contract stores?

Reply:
Done ✅